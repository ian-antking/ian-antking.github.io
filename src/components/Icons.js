import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const IconBox = styled.div`
  width: 100px;
  height: 100px;
  margin: 20px;
`;

const IconsBox = styled.div`
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
`;

const Icons = ({ icons }) => (
  <IconsBox data-testid="icons-box">
    {icons.map((icon) => (
      <IconBox data-testid="rendered-icon" key={icon.name}>
        <icon.Icon />
      </IconBox>
    ))}
  </IconsBox>
);

Icons.propTypes = {
  icons: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      Icon: PropTypes.func.isRequired,
      type: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default Icons;
