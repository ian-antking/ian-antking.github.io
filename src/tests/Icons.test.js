import React from 'react';
import { render } from '@testing-library/react';
import Icons from '../components/Icons';

describe('Icons', () => {
  const icons = [
    {
      name: 'test-name1',
      Icon: () => <div>mock-icon1</div>,
      type: 'test-type',
    },
    {
      name: 'test-name2',
      Icon: () => <div>mock-icon2</div>,
      type: 'test-type',
    },
    {
      name: 'test-name3',
      Icon: () => <div>mock-icon3</div>,
      type: 'test-type',
    },
  ];

  it('renders the correct snapshot', () => {
    const { asFragment } = render(<Icons icons={icons} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the IconsBox', () => {
    const { getByTestId } = render(<Icons icons={icons} />);

    const iconBox = getByTestId('icons-box');
    expect(iconBox).toBeInTheDocument();
  });

  it('renders the correct number of icons', () => {
    const { getAllByTestId } = render(<Icons icons={icons} />);

    const renderedIcons = getAllByTestId('rendered-icon');
    expect(renderedIcons).toHaveLength(3);
  });
});
