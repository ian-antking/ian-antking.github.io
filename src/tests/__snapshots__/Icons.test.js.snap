// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders the correct snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ qPyFg"
    data-testid="icons-box"
  >
    <div
      class="sc-AxjAm bFEEkt"
      data-testid="rendered-icon"
    >
      <div>
        mock-icon1
      </div>
    </div>
    <div
      class="sc-AxjAm bFEEkt"
      data-testid="rendered-icon"
    >
      <div>
        mock-icon2
      </div>
    </div>
    <div
      class="sc-AxjAm bFEEkt"
      data-testid="rendered-icon"
    >
      <div>
        mock-icon3
      </div>
    </div>
  </div>
</DocumentFragment>
`;
